This repository contains scripts for the bioinformatic analysis used in the article "Dissecting multitrophic interactions: the relationships among Entomophthora, their dipteran hosts, and associated bacteria."
1. MultiSPLIT.py - script used for splitting amplicon reads based on the targetted marker gene
2. LSD.py - wrapper script that merges forward and reverse reads and converts them to a FASTA format based on a PHRED score of at least 30. Then, the script dereplicates, denoises, assigns taxonomy, and clusters sequences based on their 97% similarity. This results in a table with read counts, their classification to OTUs, and their taxonomic designations across the samples.
3. QUACK.py - a decontamination script that further splits sequences based on their taxonomical assignment and provides a table for further manual processing. Reads assigned as contaminants based on their presence in negative samples or by their taxonomic labels (e.g., reads coming from mitochondria, chloroplasts, archaea) are removed, while the reads of quantification spike-ins are kept for manual bacterial abundance estimation.

   Additionally, processed data was visualised using this Processing script: Barplot_visualisation_script.pyde
